### PR TITLE
Made connection icon to be remembered by memento in ConnectionManagementService

### PR DESCRIPTION
**Pull request by geleems**
This PR makes connection icon to be remembered by memento in ConnectionManagementService so that connections can be rendered with correct icon after ADS execution.

GIF

### CODE REVIEW

File: src/sql/platform/connection/common/connectionManagementService.ts [hunk 1]
```diff
@@ -562,15 +573,20 @@ export class ConnectionManagementService extends Disposable implements IConnecti
...
+					this._mementoObj.CONNECTION_ICON_ID[connectionProfile.id] = iconId;
+					this._mementoContext.saveMemento();
 				}
 			});
 		}
 	}
 
 	public getConnectionIconId(connectionId: string): string {
-		return this._connectionIconIdCache.get(connectionId);
+		if (!connectionId || !this._mementoObj || !this._mementoObj.CONNECTION_ICON_ID) { return undefined; }
```
Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/connectionManagementService.ts]: nit: body of it statements on separate lines please (even for single lines). Easy to misread this as executing the below line instead

Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/connectionManagementService.ts]: If we for some reason return undefined does the icon fall back to the default one?

Comment (geleems) [reply src/sql/platform/connection/common/connectionManagementService.ts]: that's correct.

Comment (geleems) [reply src/sql/platform/connection/common/connectionManagementService.ts]: fixed.

**Pull request merged**